### PR TITLE
Various small fixes

### PR DESCRIPTION
**Pull request by huitseeker**
- some better option usage
- removed some unneeded toStrings
- some use of builtin functions for when applicable

### CODE REVIEW

**Comment by fnothaft** [comment]
Jenkins, test this please.

File: adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantAnnotationConverter.scala [hunk 1]
```diff
@@ -113,7 +113,7 @@ object VariantAnnotationConverter extends Serializable {
   private def createFieldMap(keys: Seq[AttrKey], schema: Schema): Map[String, (Int, Object => Object)] = {
     keys.filter(_.attrConverter != null).map(field => {
       val avroField = schema.getField(field.adamKey)
-      field.vcfKey -> (avroField.pos, field.attrConverter)
+      field.vcfKey -> ((avroField.pos, field.attrConverter))
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantAnnotationConverter.scala]: sorry, what does this do?

Comment (huitseeker) [inline adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantAnnotationConverter.scala]: The compiler is in this case auto-tupling and inserting the parentheses that I'm here making more explicit. See e.g. http://stackoverflow.com/a/29252334/47978

It's an [`ill--documented`](https://issues.scala-lang.org/browse/SI-3583) and now kind of deprecated feature.

Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantAnnotationConverter.scala]: thanks!  yeah, that shouldn't have worked before  :)

**Comment by heuermh** [comment]
LGTM
@fnothaft merge this now or wait until #906 lands in case of conflicts?

**Comment by fnothaft** [comment]
Thanks for the great changes @huitseeker! I've dropped a variety of notes inline. Thanks for making a pass over the code and cleaning up a lot of detritus. It's especially good to see a lot of the `.toString` calls go away; these came in originally due to an issue in the way we were building the `bdg-formats` schemas, as they would return UTF8 strings instead of normal Java strings and would give all sorts of weird behavior.

**Comment by fnothaft** [comment]
Thanks for the updated changes, @huitseeker! They look really good. I just had one last comment about the FASTQ exception; let me know what your thoughts are, and I think we can get this merged very soon.

**Comment by huitseeker** [comment]
Done ! Let me know what you think (and don't forget https://github.com/bigdatagenomics/utils/pull/60 )

**Comment by heuermh** [comment]
Jenkins, retest this please

**Comment by heuermh** [comment]
LGTM, thanks!

**Comment by heuermh** [comment]
Sorry, there's an error with the formatter
```none
[INFO] --- scalariform-maven-plugin:0.1.4:format (default-cli) @ adam-parent_2.10 ---
[ERROR] Error formatting /home/jenkins/workspace/ADAM-prb/HADOOP_VERSION/2.6.0/SCALAVER/2.10/SPARK_VERSION/1.4.1/label/centos/adam-core/src/test/scala/org/bdgenomics/adam/rdd/read/MDTaggingSuite.scala: java.nio.charset.MalformedInputException: Input length = 1
[ERROR] Error formatting /home/jenkins/workspace/ADAM-prb/HADOOP_VERSION/2.6.0/SCALAVER/2.10/SPARK_VERSION/1.4.1/label/centos/adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDFunctions.scala: java.nio.charset.MalformedInputException: Input length = 1
[ERROR] Error formatting /home/jenkins/workspace/ADAM-prb/HADOOP_VERSION/2.6.0/SCALAVER/2.10/SPARK_VERSION/1.4.1/label/centos/adam-cli/src/main/scala/org/bdgenomics/adam/cli/PrintADAM.scala: java.nio.charset.MalformedInputException: Input length = 1
[INFO] Modified 1 of 199 .scala files
```
I'm for removing this from the build but will probably be outvoted.

**Comment by heuermh** [comment]
@fnothaft @massie somewhat related, how do we whitelist a contributor for Jenkins?  follow up via email if necessary

**Comment by huitseeker** [comment]
@heuermh "add to whitelist"

**Comment by huitseeker** [comment]
Updated to something more compliant. Let me know how it goes for you. I'm happy to revert the last commit if it goes against conventions

**Comment by fnothaft** [comment]
Thanks @huitseeker! This looks stellar from my side. @heuermh, this is good to go from your side as well, right? If so, I will squash this down later today and merge.

**Comment by heuermh** [comment]
Jenkins, add to whitelist

**Comment by huitseeker** [comment]
Guys, I can't really do anything re: the three last formatting errors, see:

```
huitseeker➜huitseeker/tmp/adam(BDGLinting✗)» git diff origin/master -- adam-core/src/test/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDFunctionsSuite.scala
huitseeker➜huitseeker/tmp/adam(BDGLinting✗)» git diff origin/master -- adam-core/src/test/scala/org/bdgenomics/adam/rdd/read/MDTaggingSuite.scala
huitseeker➜huitseeker/tmp/adam(BDGLinting✗)» git diff origin/master -- adam-cli/src/main/scala/org/bdgenomics/adam/cli/PrintADAM.scala       [20:55:44]
huitseeker➜huitseeker/tmp/adam(BDGLinting✗)» 
```

**Comment by fnothaft** [comment]
> Guys, I can't really do anything re: the three last formatting errors, see:

Sure, I'll take a look into this.

**Comment by heuermh** [comment]
Closing in favor of #908

**Comment by fnothaft** [comment]
Thanks again @huitseeker for the great change! I squashed it down and fixed the source formatting issue that was causing the build to fail as PR #908.

**Pull request closed without merging**